src/status_im/common/standard_authentication/events.cljs

```diff
@@ -1,13 +1,101 @@
 (ns status-im.common.standard-authentication.events
   (:require
-    [utils.re-frame :as rf]))
+    [schema.core :as schema]
+    [status-im.common.standard-authentication.enter-password.view :as enter-password]
+    [status-im.common.standard-authentication.events-schema :as events-schema]
+    [taoensso.timbre :as log]
+    [utils.i18n :as i18n]
+    [utils.re-frame :as rf]
+    [utils.security.core :as security]))
 
-(rf/reg-event-fx :standard-auth/on-biometric-success
- (fn [{:keys [db]} [callback]]
-   (let [key-uid (get-in db [:profile/profile :key-uid])]
-     {:fx [[:keychain/get-user-password [key-uid callback]]]})))
+(defn authorize
+  [{:keys [db]} [args]]
+  (let [key-uid (get-in db [:profile/profile :key-uid])]
+    {:fx [[:biometric/check-if-available
+           {:key-uid    key-uid
+            :on-success #(rf/dispatch [:standard-auth/authorize-with-biometric args])
+            :on-fail    #(rf/dispatch [:standard-auth/authorize-with-password args])}]]}))
+
+(schema/=> authorize events-schema/?authorize)
+(rf/reg-event-fx :standard-auth/authorize authorize)
+
+(defn authorize-with-biometric
+  [_ [{:keys [on-auth-success on-auth-fail] :as args}]]
+  (let [args-with-biometric-btn
+        (assoc args
+               :on-press-biometric
+               #(rf/dispatch [:standard-auth/authorize-with-biometric args]))]
+    {:fx [[:dispatch [:dismiss-keyboard]]
+          [:dispatch
+           [:biometric/authenticate
+            {:prompt-message (i18n/label :t/biometric-auth-confirm-message)
+             :on-cancel      #(rf/dispatch [:standard-auth/authorize-with-password
+                                            args-with-biometric-btn])
+             :on-success     #(rf/dispatch [:standard-auth/on-biometric-success on-auth-success])
+             :on-fail        #(rf/dispatch [:standard-auth/on-biometric-fail on-auth-fail %])}]]]}))
+
+(schema/=> authorize-with-biometric events-schema/?authorize-with-biometric)
+(rf/reg-event-fx :standard-auth/authorize-with-biometric authorize-with-biometric)
+
+(defn on-biometric-success
+  [{:keys [db]} [on-auth-success]]
+  (let [key-uid (get-in db [:profile/profile :key-uid])]
+    {:fx [[:keychain/get-user-password [key-uid on-auth-success]]
+          [:dispatch [:standard-auth/reset-login-password]]]}))
+
+(schema/=> on-biometric-success events-schema/?on-biometric-success)
+(rf/reg-event-fx :standard-auth/on-biometric-success on-biometric-success)
+
+(defn on-biometric-fail
+  [_ [on-auth-fail error]]
+  (when on-auth-fail
+    (on-auth-fail error))
```

Comment:
This is odd in the event handler. There's no return value, so it's assumed to be a side-effect, but event handlers should only describe side-effects, not perform them. In re-frame we aspire to separate planning from execution.

Reply:
This file was added accidentally. I removed the entire file

Reply:
Okay, thanks @mmilad75